src/main/java/br/com/deveficiente/calendario/validator/PeriodoEventoValidator.java

```diff
@@ -0,0 +1,27 @@
+package br.com.deveficiente.calendario.validator;
+
+import br.com.deveficiente.calendario.controller.form.NovoEventoForm;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import java.time.LocalDateTime;
+
+public class PeriodoEventoValidator implements Validator {
+
+    @Override
+    public boolean supports(final Class<?> clazz) {
+        return NovoEventoForm.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public void validate(final Object target, final Errors errors) {
+        final NovoEventoForm novoEventoForm = (NovoEventoForm) target;
+        final LocalDateTime inico = novoEventoForm.getInico();
+        final LocalDateTime termino = novoEventoForm.getTermino();
+
+        if (!termino.isAfter(inico)) {
```

Comment:
Esse código está bem de boa. Por outro lado, inicio e termino são atributos do form. Em vez fazer a lógica em cima do retorno dos dados do mesmo objeto, você pode criar um método que já te retorne o boolean em relação ao término depois do começo. Em geral, maximizar o uso do **this** leva a deixar o estado perto do comportamento e com isso o código fica mais coeso. E coesão tende a facilitar o entendimento :).

Reply:
Boa! Criei um método `isTerminoAfterInicio()` no form, mas precisei usar `@JsonAutoDetect(isGetterVisibility = JsonAutoDetect.Visibility.NONE)` pra ele não aparecer no Swagger. É isso mesmo?

Reply:
Ele pega por conta do padrão do nome, que começa com is. Da para mandar um @JsonIgnore em cima do método ou trocar o nome :P. Nessas classes de DTO ta tudo bem ser um pouco mais flexível no design.